editor/src/clj/editor/editor_extensions.clj

```diff
@@ -3,47 +3,89 @@
 ;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
 ;; Licensed under the Defold License version 1.0 (the "License"); you may not use
 ;; this file except in compliance with the License.
-;; 
+;;
 ;; You may obtain a copy of the License, together with FAQs at
 ;; https://www.defold.com/license
-;; 
+;;
 ;; Unless required by applicable law or agreed to in writing, software distributed
 ;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 ;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
 ;; specific language governing permissions and limitations under the License.
 
 (ns editor.editor-extensions
-  (:require [clojure.java.io :as io]
-            [clojure.spec.alpha :as s]
+  (:require [clojure.set :as set]
             [clojure.stacktrace :as stacktrace]
-            [clojure.string :as string]
             [dynamo.graph :as g]
             [editor.code.data :as data]
-            [editor.code.util :as code.util]
             [editor.console :as console]
             [editor.defold-project :as project]
+            [editor.editor-extensions.actions :as actions]
+            [editor.editor-extensions.commands :as commands]
+            [editor.editor-extensions.error-handling :as error-handling]
+            [editor.editor-extensions.graph :as graph]
+            [editor.editor-extensions.runtime :as rt]
+            [editor.editor-extensions.validation :as validation]
             [editor.error-reporting :as error-reporting]
             [editor.fs :as fs]
+            [editor.future :as future]
             [editor.graph-util :as gu]
             [editor.handler :as handler]
             [editor.lsp :as lsp]
-            [editor.luart :as luart]
-            [editor.outline :as outline]
-            [editor.process :as process]
-            [editor.properties :as properties]
-            [editor.resource :as resource]
+            [editor.lsp.async :as lsp.async]
             [editor.system :as system]
-            [editor.types :as types]
             [editor.util :as util]
             [editor.workspace :as workspace])
-  (:import [clojure.lang MultiFn]
-           [com.defold.editor.luart SearchPath]
-           [com.dynamo.bob Platform]
-           [java.io File]
-           [java.nio.file FileAlreadyExistsException NotDirectoryException Path]
-           [org.luaj.vm2 LuaError LuaFunction LuaValue Prototype]))
+  (:import [com.dynamo.bob Platform]
+           [java.nio.file FileAlreadyExistsException Files LinkOption NotDirectoryException Path]
+           [org.luaj.vm2 LuaError Prototype]))
```

Comment:
Many many changes in this file, that can be summarized as follows:
- split some code to separate namespaces (related to actions, commands, error handling, validation and graph interactions) — see corresponding `.clj` files in `editor_extensions` folder.
- refactored API functions to use non-blocking, suspendable processing.
- added more docstrings